File: src/main/kotlin/lec07/Lec07Main.java
package lec07;

import org.jetbrains.annotations.NotNull;

public class Lec07Main {
    public static void main(String[] args) {
    }

    private int parseIntThrow(@NotNull String str) {
        try {
            return Integer.parseInt(str);
        } catch (NumberFormatException e) {
            throw new NumberFormatException(String.format("주어진 문자열 %s 는 문자열이 아닙니다", str));
        }
    }
}

File: src/main/kotlin/lec07/Lec07Main.kt
package lec07

import kotlin.NumberFormatException

fun parseIntThrow(str: String): Int {
    // try-catch문 또한 if-else 문처럼 Expression이기 때문에 바로 return 할 수 있다.
    return try {
        str.toInt()
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("주어진 문자열 $str 은 문자열이 아닙니다.")
    }
}